components/announcement_bar/default_announcement_bar/announcement_bar.tsx

```diff
@@ -27,8 +27,9 @@ type Props = {
     type: string;
     message: React.ReactNode;
     handleClose?: (e?: any) => void;
+    showModal?: boolean;
```

Comment:
Should we rename this to `showButton` as well for consistency?

Reply:
showModal is used by `configuration_bar.tsx` as a boolean on whether or not to show the _link_ in the banner (these will be for older banners). I think this is where the typescript conversion tripped up - javascript was a little more lenient on the type of that prop and would use it as a boolean or a function depending on the context. So I've left this in to support that case and then added the new prop for our banners

Reply:
Have we looked at refactoring what those other components are passing in? Probably OOS for this PR but i'm just curious what the effort would be, if it's even possible.